Citadel 0.0.8

AaronDewes opened the pull request:
This update updates Citadel's backend to have most of the features planned for 0.1.0, but does not yet utilze them, except most of the new app system.

AaronDewes committed:
App system cleanups (#51)

* Add app cli to docker-compose.yml

* Remove app.yml v1

* Add missing import

* More cleanups

* Another missing import

* Add mount for apps

* Remove more

AaronDewes committed:
Implement quick updates (#56)

Co-authored-by: nolim1t - f6287b82CC84bcbd <nolim1t@users.noreply.github.com>
Co-authored-by: Philipp Walter <philippwalter@pm.me>

AaronDewes committed:
Implement quick updates (#56)

Co-authored-by: nolim1t - f6287b82CC84bcbd <nolim1t@users.noreply.github.com>
Co-authored-by: Philipp Walter <philippwalter@pm.me>

AaronDewes committed:
Citadel 0.1.0 Preview 1 (#67)

AaronDewes committed:
Fix: 0.1.0 update (#68)

AaronDewes committed:
Update info.json

AaronDewes committed:
Preprocess app.yml.jinja files (#70)

* Preprocess app.yml.jinja files

* Fix preprocess

* Update app cli

* Try to fix permissions

* Actually fix permissions

AaronDewes committed:
Update sources.list (#71)

AaronDewes committed:
Preview 3

AaronDewes committed:
Remove unused function

AaronDewes committed:
Update all dependencies

AaronDewes committed:
Add basic support for virtual apps

AaronDewes committed:
Remove services argument for app compiler

AaronDewes committed:
Remove another unused argument

AaronDewes committed:
Citadel 0.0.8

AaronDewes committed:
Use stable app repos

AaronDewes committed:
Fix a typo

AaronDewes committed:
Update docker-compose.yml to match dependencies.yml

pwltr commented on app/lib/metadata.py:
```diff
@@ -4,10 +4,10 @@
-from lib.composegenerator.next.stage1 import createCleanConfigFromV3
 from lib.composegenerator.v2.networking import getMainContainer
-from lib.composegenerator.v1.networking import getFreePort
+from lib.composegenerator.shared.networking import assignIpV4
```
I don't see this file anywhere, could it be missing?

AaronDewes replied:
This should be fixed now

AaronDewes committed:
Remove reference to non-existent var

AaronDewes committed:
Update compose

AaronDewes committed:
Better separation of composegenertor v2 and v3

AaronDewes committed:
Mark 0.1.5 as removal version for app.yml v2 and v3

AaronDewes committed:
Fix imports

AaronDewes committed:
Remove another unused import

AaronDewes committed:
Move utils to shared

AaronDewes committed:
Fix app port reassignment & race condition when writing to registry.json

AaronDewes committed:
Reduce service dependencies

- Nginx can start before manager, if the dashboard is accessible, we can already show a loading anmimation, meaning we'll have faster boot times
- There is no need for middleware to depend on manager or to wait for it to start. Middleware never connects to manager and runs completely independently

This should increase boot speed and helps making the services more independent (for quick updates)

pwltr commented on docker-compose.yml:
```diff
@@ -162,13 +160,12 @@ services:
     - bitcoin
     - lightning
     - redis
-    command: sh -c "./wait-for-manager.sh $MANAGER_IP && ./start.sh"
```
Removing this causes issues with JWT. As the auth code is written now the manager needs to be running before middleware otherwise the JWT is invalid from the manager's perspective.

AaronDewes committed:
Re-add waiting for manager
